bot/seasons/valentines/be_my_valentine.py

```diff
@@ -0,0 +1,228 @@
+import logging
+import random
+import typing
+from json import load
+from pathlib import Path
+
+import discord
+from discord.ext import commands
+from discord.ext.commands.cooldowns import BucketType
+
+from bot.constants import Colours, Lovefest, Roles
+from bot.decorators import with_role
+
+log = logging.getLogger(__name__)
+
+HEART_EMOJIS = [":heart:", ":gift_heart:", ":revolving_hearts:", ":sparkling_heart:", ":two_hearts:"]
+EMOJI_1 = random.choice(HEART_EMOJIS)
+EMOJI_2 = random.choice(HEART_EMOJIS)
+USER_LOVEFEST = []
+JSON_FILE = open(Path('bot', 'resources', 'VALENTINES', 'bemyvalentine_valentines.json'), 'r', encoding="utf8")
+VALENTINES = load(JSON_FILE)
+
+
+class BeMyValentine:
+    """
+    A cog that sends VALENTINES to other users !
+    """
+    id = Lovefest.role_id
+
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.command(name="lovefest")
+    async def add_role(self, ctx):
+        """
+        This command adds people to the lovefest role.
+        """
+        user = ctx.author
+        role = discord.utils.get(ctx.guild.roles, id=Lovefest.role_id)
+        if id not in [role.id for role in ctx.message.author.roles]:
+            USER_LOVEFEST.append(ctx.author)
+            await user.add_roles(role)
+            await ctx.send("The Lovefest role has been added !")
+        else:
+            await ctx.send("You already have the role !")
+
+    @with_role(Roles.moderator)
+    @commands.command(name='refreshlovefest')
+    async def refresh_user_lovefestlist(self, ctx):
```

Comment:
Can you provide some information on why this is needed as a command?

Reply:
well when people use .bemyvalentine (without providing a user to send a valentine to) ,then a random user having the lovefest role will be selected.So basically i have a list of people who have the lovefest role called USER_LOVEFEST.When people use the .lovefest command to get the lovefest role,then their name will get appended to this list.
But, INCASE the bot goes offline and comes back online,the list is lost, so thats why this .refreshlovefest

Reply:
I don't see why this list of folks with the role is necessary. Checking [`Member.roles`](https://discordpy.readthedocs.io/en/rewrite/api.html#discord.Member.roles) (probably with a helper method) accomplishes the same goal without needing to worry about maintaining a cache of users with the role in memory.

Reply:
so i should use it whenever someone runs the command ? like instantly?

Reply:
like make a method that does the same thing but only when its needed?
like when someone uses the command without providing a users name